### PR TITLE
CIAB: Add static subnet

### PR DESCRIPTION
**Pull request by Shihta**
#### What does this PR do?

Since docker-compose will randomly create a subnet and it has a chance to
conflict with your network environment.
So fix it would be better.

#### Which TC components are affected by this PR?

- [ ] Documentation
- [ ] Grove
- [ ] Traffic Analytics
- [ ] Traffic Monitor
- [ ] Traffic Ops
- [ ] Traffic Ops ORT
- [ ] Traffic Portal
- [ ] Traffic Router
- [ ] Traffic Stats
- [ ] Traffic Vault
- [x] Other CIAB

#### What is the best way to verify this PR?

Using `docker-compose up` and check it status

#### Check all that apply

- [ ] This PR includes tests
- [x] This PR includes documentation updates
- [ ] This PR includes an update to CHANGELOG.md
- [ ] This PR includes all required license headers
- [ ] This PR includes a database migration (ensure that migration sequence is correct)
- [ ] This PR fixes a serious security flaw. Read more: [www.apache.org/security](http://www.apache.org/security/)

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: docs/source/admin/quick_howto/ciab.rst [hunk 1]
```diff
@@ -310,3 +310,15 @@ Dante's socks proxy is an optional container that can be used to provide browser
 #. Once the CDN-in-a-box stack has started, use the aforementioned browser to access traffic portal via the socks proxy on the docker host.
 
 .. seealso:: `The official Docker Compose documentation CLI reference <https://docs.docker.com/compose/reference/>`_ for complete instructions on how to pass service definition files to the ``docker-compose`` executable.
+
+Static Subnet
+-------------
+Since docker-compose will randomly create a subnet and it has a chance to conflict with your network environment, using static subnet is a good choice.
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: `docker-compose` is a command, should be wrapped in double-backticks. Or, if you're feeling like going above and beyond the call of duty, make it a manpage reference:

```rst
:manpage:`docker-compose(1)`
```

File: docs/source/admin/quick_howto/ciab.rst [hunk 2]
```diff
@@ -310,3 +310,15 @@ Dante's socks proxy is an optional container that can be used to provide browser
...
+
+.. code-block:: shell
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: formatting looks fine, but this needs a caption so it's linkable.

Comment (ocket8888) [reply docs/source/admin/quick_howto/ciab.rst]: No, that's a reference label. If you look in the VNC container section you'll see:

```rst
	.. code-block:: shell
		:caption: CIAB Startup Using Bash Alias

		# From infrastructure/cdn-in-a-box
		alias mydc="docker-compose -f $PWD/docker-compose.yml -f $PWD/optional/docker-compose.vnc.yml"
		docker volume prune -f
		mydc build
		mydc kill
		mydc rm -fv
		mydc up
```
the `:caption:` option specifies the code block caption. There are other code-blocks without captions, but they shouldn't be. (You of course only need to add captions to your contribution; we'll sort the others out in time)

Comment (Shihta) [reply docs/source/admin/quick_howto/ciab.rst]: Thank you for your explanation. I have fixed it

File: docs/source/admin/quick_howto/ciab.rst [hunk 3]
```diff
@@ -310,3 +310,16 @@ Dante's socks proxy is an optional container that can be used to provide browser
 #. Once the CDN-in-a-box stack has started, use the aforementioned browser to access traffic portal via the socks proxy on the docker host.
 
 .. seealso:: `The official Docker Compose documentation CLI reference <https://docs.docker.com/compose/reference/>`_ for complete instructions on how to pass service definition files to the ``docker-compose`` executable.
+
+Static Subnet
+-------------
+Since ``docker-compose`` will randomly create a subnet and it has a chance to conflict with your network environment, using static subnet is a good choice.
+
+.. code-block:: shell
+	:caption: CIAB Startup Using Bash Alias
```
Comment (ocket8888) [inline docs/source/admin/quick_howto/ciab.rst]: Captions should be unique. This does more than just "start up CDN-in-a-Box", it's doing it with a static subnet.

Comment (Shihta) [reply docs/source/admin/quick_howto/ciab.rst]: Sorry that there are tons of duplicate caption so I didn't aware that. I have fixed it.
Thanks!

**Comment by JBevillC** [comment]
@rob05c @ocket8888  I've tested this successfully.  The original CiaB static subnets are used if the optional `docker-compose.static-subnets.yml` is specific via an additional `-f` flag to docker-compose.   Tested with the new exposed-ports changes in #3270.  Other than the documentation change that is requested, this PR is ready to be merged.

My testing procedure:

```
# From ./infrastructure/cdn-in-a-box
$ alias mydc="docker-compose "` \
        `"-f $PWD/docker-compose.yml "` \
        `"-f $PWD/docker-compose.expose-ports.yml "` \
        `"-f $PWD/optional/docker-compose.socksproxy.yml "` \
        `"-f $PWD/optional/docker-compose.socksproxy.expose-ports.yml "` \
        `"-f $PWD/optional/docker-compose.vnc.yml "` \
        `"-f $PWD/optional/docker-compose.vnc.expose-ports.yml "` \
        `"-f $PWD/optional/docker-compose.static-subnet.yml "  # Use static subnet 172.16.239.0/16
$ make very-clean && make -j 7  # Rebuild RPMS
$ docker network prune -f           # Nuke old stale CiaB networks
$ mydc build                                  # Rebuild any stale docker containers
$ mydc kill && mydc rm -fv           # Nuke old CiaB containers
$ docker volume prune -f             
$ mydc up -d
```

Inspecting the docker network, I observe that the static subnet is indeed being created correctly:
```
$ docker network inspect cdn-in-a-box_default | jq '.[0].IPAM'
```
Output:
```
{
  "Driver": "default",
  "Options": null,
  "Config": [
    {
      "Subnet": "172.16.239.0/24"
    },
    {
      "Subnet": "fc01:9400:1000:8::/64"
    }
  ]
}
```
Make sure exposed-ports are correct due to changes from PR #3270:
```
$ mydc ps
```
Output:
```
$ mydc ps
                   Name                                  Command               State                                              Ports                                          
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
cdn-in-a-box_db_1_60d2d57af8d4                docker-entrypoint.sh postgres    Up       0.0.0.0:5432->5432/tcp                                                                   
cdn-in-a-box_dns_1_44b696605988               /sbin/entrypoint.sh /usr/s ...   Up       0.0.0.0:9353->53/tcp, 53/udp                                                             
cdn-in-a-box_edge_1_d834ba5d0d5f              /bin/sh -c /run.sh               Up       0.0.0.0:9000->80/tcp                                                                     
cdn-in-a-box_enroller_1_38ab16b18eda          /bin/sh -c /run.sh               Up                                                                                                
cdn-in-a-box_mid_1_508934d3a10e               /bin/sh -c /run.sh               Up       0.0.0.0:9100->80/tcp                                                                     
cdn-in-a-box_origin_1_d351d0fba750            /bin/sh -c /run.sh               Up       0.0.0.0:9200->80/tcp                                                                     
cdn-in-a-box_socksproxy_1_3cb68ba771a2        /run.sh                          Up       0.0.0.0:9080->1080/tcp                                                                   
cdn-in-a-box_tccache_1_9c2444ef8530           /bin/sh -c exit                  Exit 0                                                                                            
cdn-in-a-box_trafficmonitor_1_935d4f6af9b9    /bin/sh -c /run.sh               Up       0.0.0.0:80->80/tcp                                                                       
cdn-in-a-box_trafficops-perl_1_f268b5c61e03   /bin/sh -c /run.sh               Up       0.0.0.0:60443->443/tcp                                                                   
cdn-in-a-box_trafficops_1_2619467f51ed        /bin/sh -c /run-go.sh            Up       0.0.0.0:6443->443/tcp                                                                    
cdn-in-a-box_trafficportal_1_eedc34486446     /bin/sh -c /run.sh               Up       0.0.0.0:443->443/tcp                                                                     
cdn-in-a-box_trafficrouter_1_f72b599c1bbc     /bin/sh -c /run.sh               Up       0.0.0.0:3333->3333/tcp, 0.0.0.0:3443->443/tcp, 0.0.0.0:3053->53/tcp, 0.0.0.0:3080->80/tcp
cdn-in-a-box_trafficvault_1_c0ca89fb3c82      /bin/sh -c /run.sh               Up       0.0.0.0:8087->8087/tcp, 0.0.0.0:8098->8098/tcp                                           
cdn-in-a-box_vnc_1_cb13fee16337               /bin/sh -c /run.sh               Up       0.0.0.0:5909->5909/tcp
```

**Comment by ocket8888** [comment]
With this change, if I'm in the CiaB directory and do `docker-compose up` with no specified optional configuration files, has anything changed?

**Comment by JBevillC** [comment]
@ocket8888 Correct.  Nothing has changed.

**Comment by JBevillC** [comment]
The static subnet network configuration is entirely optional.

**Comment by ocket8888** [comment]
In that case, your approval is enough for me :+1:

**Pull request merged**